### PR TITLE
feat(chat): improve auto follow cursor behavior

### PR DESCRIPTION
**Pull request by taketwo**
Auto follow cursor is generally a convenient and desirable feature. However, it gets annoying when the model is streaming a very long response spanning more vertical space than the chat window has. In such cases the user might want to read the response from the beginning at a human pace, yet auto follow keeps pushing forward at robot speed.

This PR improves auto follow cursor behavior by conditioning it on the current cursor position. If it is at the last line, auto-following happens. Otherwise, the cursor stays where it is. Effectively, this means that the user can prevent auto-following from happening by simply moving the cursor up.

### CODE REVIEW

File: lua/CopilotChat/chat.lua [hunk 1]
```diff
@@ -129,6 +131,16 @@ function Chat:append(str)
     self.spinner:start()
   end
 
+  -- Decide if we should follow cursor after appending text.
+  -- Note that winnr may be nil if the chat window is not open yet.
+  local should_follow_cursor = self.auto_follow_cursor
+  if self.auto_follow_cursor and self.winnr then
```
Comment (deathbeam) [inline lua/CopilotChat/chat.lua:137]: this check should probably be behind self:visible() instead of just checking if winnr is set.

**Review by deathbeam — CHANGES_REQUESTED** [review]

File: lua/CopilotChat/chat.lua [hunk 2]
```diff
@@ -2,6 +2,7 @@
 ---@field bufnr number
 ---@field winnr number
 ---@field separator string
+---@field auto_follow_cursor boolean
```
Comment (deathbeam) [inline lua/CopilotChat/chat.lua:5]: dont need to add this here I think, lsp should know by default when working in this file, and when working outside we dont want this field "exposed" anyway

**Review by deathbeam — CHANGES_REQUESTED** [review]

**Review by taketwo — COMMENTED** [review]
Thanks for the review, makes sense, I've pushed a fixup commit.

**Review by deathbeam — APPROVED** [review]
Looks good, thanks!

**Comment by deathbeam** [comment]
@all-contributors add @taketwo for code

**Pull request merged**